### ISSUE TITLE
Watching `__dirname` and not what I set `path` to

### ISSUE THREAD

**Issue by johnletey** [yan-foto/electron-reload#61]
## What is the problem
When I set the path to watch as a different path other than `__dirname`, it doesn't seem to work ... what I mean is that it seems to watch changes only in `__dirname`, and not the path that I specified

## How to reproduce
- Clone [blocks/desktop](https://github.com/blocks/desktop)
- `yarn`
- `yarn add -D electron-reload`
- Add this code to `main/index.js`:

```js
require('electron-reload')(path.join(__dirname, '..', 'src'), {
  electron: path.join(__dirname, '..', 'node_modules', '.bin', 'electron')
})
```

- `yarn start`
- Change line 83 in `src/index.js` to:

```diff
- <span sx={{ ml: 2 }}>Back</span>
+ <span sx={{ ml: 2 }}>Testing</span>
```

- Navigate to some MD or MDX file in your filesystem
- It will show up as `<- Back` and not as `<- Testing`! For example:

<table>
    <tr>
       <td>Before</td>
       <td>After</td>
   </tr>
   <tr>
       <td><img src="https://user-images.githubusercontent.com/30328854/61643752-640d1c80-ac9b-11e9-837b-517050d98589.jpg" /></td>
       <td><img src="https://user-images.githubusercontent.com/30328854/61643809-82731800-ac9b-11e9-9361-b104527abc5c.jpg" /></td>
    </tr>
</table>

## Additional information
OS: macOS 10.14.4 (18E226) ... Mojave
node: 10.15.2
yarn: 1.15.2
electron: 5.0.7

**Comment by yan-foto** [comment]
I don't know what you mean by "it doesn't seem to work". It would be best to provide a generic "how to reproduce" instruction instead of pointing to an external project.

Nonetheless, I suppose that you have problems with hard resets and don't require `electron-reload` from your main file. If this is the case, you should take a look at this line:

https://github.com/yan-foto/electron-reload/blob/master/main.js#L12

But this is just a wild guess, it might be something else. It might the case that you transpile your JSX to JS in some other folder and watch the sources instead of the transpiled files. I cannot debug your project.

**Comment by johnletey** [comment]
> I suppose that you have problems with hard resets and don't require `electron-reload` from your main file.

That is exactly what's going on

> But this is just a wild guess, it might be something else. It might the case that you transpile your JSX to JS in some other folder and watch the sources instead of the transpiled files. I cannot debug your project.

Thank you so much for taking the time ... I will continue to debug and update you on what I find!

**Comment by yan-foto** [comment]
Yes please let me know. We made a change in the last release and made the assumption that the main file requires `electron-reload`. Maybe it doesn't make sense for specific projects and we need to revert back.

**Comment by johnletey** [comment]
> Maybe it doesn't make sense for specific projects and we need to revert back.

You should ... I believe that would fix my problem here!

**Comment by yan-foto** [comment]
Easier said than done! There was a reason why we did it (see #58 ) and I'd need a good reason to make new changes.

Why don't you require `electron-reload` from your main file?

**Comment by johnletey** [comment]
> Why don't you require `electron-reload` from your main file?

Because I get a blank screen:

![Shot 2019-07-22 at 16 44 35](https://user-images.githubusercontent.com/30328854/61646029-10510200-aca0-11e9-98f3-21bcfb4cf5bd.jpg)

Note that I added this code:

```js
require('electron-reload')(__dirname, {
  electron: path.join(__dirname, '..', 'node_modules', '.bin', 'electron')
})
```

**Comment by yan-foto** [comment]
Now you just need to find what goes wrong!

**Comment by johnletey** [comment]
I think what is going wrong is that `src/index.js` doesn't actually do anything related to electron ... but `main/index.js` does

When I put the code in `src/index.js`, it shows up blank
When I put the code in `main/index.js`, it doesn't watch `src/*`

**Comment by yan-foto** [comment]
Do me a favor, change the source code of `electron-reload` in your `node_modules` directory and add this to line 13 of `main.js` and let me know what it shows:

`console.log(mainFile)`

**Comment by yan-foto** [comment]
You are talking about two different things: hard and soft resets. Soft resets happen when files are changed in the directory that you passed to `electron-reload`. It just reloads all `BrowserWindow`s. Hard reset restarts electron completely when your main file (as defined in `package.json`) changes. I'm still not sure what's the situation here.

**Comment by johnletey** [comment]
> Do me a favor, change the source code of `electron-reload` in your `node_modules` directory and add this to line 13 of `main.js` and let me know what it shows:

Done ... but nothing prints:

![Shot 2019-07-22 at 17 02 53](https://user-images.githubusercontent.com/30328854/61646619-9bcb9280-aca2-11e9-9d31-598f39f8e89e.jpg)

> I'm still not sure what's the situation here.

The way you described it, I want to reload all `BrowserWindow`s

FYI, `main/index.js` is set as the main file in `package.json`

**Comment by yan-foto** [comment]
Does the structure of transpiled files reflect the structure of your current source (i.e. before and after babel)

**Comment by johnletey** [comment]
> Does the structure of transpiled files reflect the structure of your current source

Where would I look for this exactly?

**Comment by johnletey** [comment]
FYI, when I add the code to `main/index.js` (the main file) ... the `console.log` command prints "/Users/john/Desktop/desktop/main/index.js" ... so `main/index.js`

But it doesn't print anything for `src/index.js`

**Comment by yan-foto** [comment]
I am really sorry, but I think this *ping-pong debugging* is not going to help us. You are using a number of tools (babel, yarn, etc.) which transform your sources and probably restructure your project after building it. I suggest that you take a step back and try to figure out what is happening in the background.

For soft resets the path that you pass to `electron-reload` must be the same as where your transpiled files reside. That is the only thing that matter! You can easily find out if its the case by printing out variables and comparing it with whats going on in your project.

**Comment by johnletey** [comment]
Thanks for all the help @yan-foto!

I think I've made some progress! When I put

```js
require('electron-reload')(path.join(__dirname, '..', 'src'), {
  electron: path.join(__dirname, '..', 'node_modules', '.bin', 'electron')
})
```

into `main/index.js` (the main file) and then edit something in `src/index.js`, the app reloads ... but the changes don't appear

Is that a little bit more meaningful?

**Comment by yan-foto** [comment]
Yes! I know what happens! We should've known it earlier. When you start with `yarn` it initiates babel, which transpiles your files. `electron-reload` does not do that! What you'd need is some kind of event triggering, which is yet to be implemented. It would also solve #35 and is something that I meant to implement for some time now.

**Comment by johnletey** [comment]
I see ... thanks for the help @yan-foto! I'll keep an eye on #35 and see if I can open a PR

**Comment by yan-foto** [comment]
Please leave this open. I'll use it as a cue for implementing the events.

**Comment by dough654** [comment]
> You are talking about two different things: hard and soft resets. Soft resets happen when files are changed in the directory that you passed to `electron-reload`. It just reloads all `BrowserWindow`s. Hard reset restarts electron completely when your main file (as defined in `package.json`) changes. I'm still not sure what's the situation here.

I know this is kind of late, and maybe it should be its' own feature request, but have you really thought through the usefulness of that feature? Shouldn't the process be watching for changes in the main file, but also any dependencies of the main file? If I import a module in to my main file, I would likely want my electron app restarted then as well. That's the scenario I find myself in. Just a thought.

**Comment by yan-foto** [comment]
@dough654 that makes perfect sense. Could you please open a new issue for discussions?
